sharding/data_modeling.rst

```diff
@@ -0,0 +1,222 @@
+
+.. code-block:: postgres
+
+  INSERT INTO daily_page_views (day, page_id, view_count)
+  SELECT view_time::date AS day, page_id, count(*) AS view_count
+  FROM page_views
+  WHERE view_time >= $1
+  GROUP BY view_time::date, page_id
+  ON CONFLICT (day, page_id) DO UPDATE SET
+    view_count = daily_page_views.view_count + EXCLUDED.view_count;
+
+Updatable Large Table
+---------------------
+
+(Device table that has characteristics that get updated. Sharded by device id.)
+
+Behavioral Analytics
+--------------------
+
+Whereas the previous examples dealt with a single events table (possibly augmented with precomputed rollups), this example uses two main tables: users and their events. Tracking user behavior is another common Citus use case. In particular consider Wikipedia editors and their edits:
+
+.. code-block:: postgres
+
+  CREATE TABLE wikipedia_editors (
+    editor TEXT UNIQUE,
+    bot BOOLEAN,
+
+    edit_count INT,
+    added_chars INT,
+    removed_chars INT,
+
+    first_seen TIMESTAMPTZ,
+    last_seen TIMESTAMPTZ
+  );
+
+  CREATE TABLE wikipedia_changes (
+    editor TEXT,
+    time TIMESTAMP WITH TIME ZONE,
+
+    wiki TEXT,
+    title TEXT,
+
+    comment TEXT,
+    minor BOOLEAN,
+    type TEXT,
+
+    old_length INT,
+    new_length INT
+  );
+
+These tables can be populated by the Wikipedia API, and we can distribute them in Citus by the :code:`editor` column. Notice that this is a text column. Citus' hash distribution uses PostgreSQL hashing which supports a number of data types.
+
+A co-located JOIN between editors and changes allows aggregates not only by user, but by properties of a user. For instance we can count the difference between the number of newly created pages by bot vs human. The grouping and counting is performed on worker nodes in parallel and the final results are merged on the coordinator node.
+
+.. code-block:: postgres
+
+  SELECT bot, count(*) AS pages_created
+  FROM wikipedia_changes c,
+       wikipedia_editors e
+  WHERE c.editor = e.editor
+    AND type = 'new'
+  GROUP BY bot;
+
+Star Schema
+-----------
+
+We've already seen how every row in a distributed table is stored on a shard. However for small tables there is a trick to achieve a kind of universal colocation. We can choose to place all its rows into a single shard but replicate that shard to every worker node. It introduces storage and update costs of course, but this can be more than counterbalanced by the performance gains of read queries.
+
+We call tables replicated to all nodes *reference tables.* They usually provide metadata about items in a larger table and are reminiscent of what data warehousing calls dimension tables. For example, suppose we have a large table of phone calls:
+
+.. code-block:: postgres
```

Comment:
@ozgune : I agree. Once we have complete reference table implementation in v6.1 (and get some input on its applicability), we can re-introduce this example.

I would however, still prefer to keep the text sections just to let people know that they can do something like this with Citus.